data-solutions/data-platform-foundations/datasource/main.tf

```diff
@@ -16,4 +16,208 @@
 
 locals {
   module_version = "4.2.0"
+
+  landing_pubsub = merge({
+    for k, v in var.landing_pubsub :
+    k => {
+      name          = v.name
+      subscriptions = v.subscriptions
+      subscription_iam = merge({
+        for s_k, s_v in v.subscription_iam :
+        s_k => merge(s_v, { "roles/pubsub.subscriber" : ["serviceAccount:${module.transformation-default-service-accounts.email}"] })
+      })
+    }
+  })
+}
+
+###############################################################################
+#                                 Projects                                    #
+###############################################################################
+module "project-datamart" {
+  source         = "../../../modules/project"
+  name           = var.project_ids.datamart
+  project_create = false
+}
+
+module "project-dwh" {
+  source         = "../../../modules/project"
+  name           = var.project_ids.dwh
+  project_create = false
+}
+
+module "project-landing" {
+  source         = "../../../modules/project"
+  name           = var.project_ids.landing
+  project_create = false
+}
+
+module "project-services" {
+  source         = "../../../modules/project"
+  name           = var.project_ids.services
+  project_create = false
+}
+
+module "project-transformation" {
+  source         = "../../../modules/project"
+  name           = var.project_ids.transformation
+  project_create = false
+}
+
+###############################################################################
+#                                   IAM                                       #
+###############################################################################
+
+module "datamart-default-service-accounts" {
+  source     = "../../../modules/iam-service-account"
+  project_id = var.project_ids.datamart
+  name       = var.project_service_account.datamart
+
+  iam_project_roles = {
+    "${var.project_ids.datamart}" = [
```

Comment:
I would refer to the project ID with a link to the module, not the variable. Please update also other elements.

Example: `module.project-datamart`.

Reply:
actually, do we need the `project` module?